File: backend/app/schemas/comparison.py
from pydantic import BaseModel
from typing import Literal


class Mismatch(BaseModel):
    id: str
    type: Literal["chapter", "heading", "paragraph", "table", "page", "word_count"]
    docxRef: str | None = None
    pdfRef: str | None = None
    message: str
    chapterIndex: int | None = None
    blockId: str | None = None


class ComparisonResult(BaseModel):
    documentId: str
    chapterCountMatch: bool
    headingMatch: bool
    paragraphCountMatch: bool
    tableCountMatch: bool
    pageCountMatch: bool
    wordCountMatch: bool
    mismatches: list[Mismatch]
    docxChapterCount: int
    pdfChapterCount: int
    docxWordCount: int
    pdfWordCount: int

File: frontend/src/types/comparison.ts
export interface Mismatch {
  id: string;
  type: 'chapter' | 'heading' | 'paragraph' | 'table' | 'page' | 'word_count';
  docxRef?: string;
  pdfRef?: string;
  message: string;
  chapterIndex?: number;
  blockId?: string;
}

export interface ComparisonResult {
  documentId: string;
  chapterCountMatch: boolean;
  headingMatch: boolean;
  paragraphCountMatch: boolean;
  tableCountMatch: boolean;
  pageCountMatch: boolean;
  wordCountMatch: boolean;
  mismatches: Mismatch[];
  docxChapterCount: number;
  pdfChapterCount: number;
  docxWordCount: number;
  pdfWordCount: number;
}
